doc/contributing/style_guides/markdown_usage_style.md

```diff
@@ -37,277 +39,141 @@ No style guide is perfect.
-
-```
-## Description
-
-Description of the IP.
-```
-The Compatibility information will allow device driver writers to identify existing code that can be used directly or with minor changes.
-
-_This section is primarily of interest to software engineers._
-
-
-```
-## Compatability
-
-Notes on if the IP register interface is compatible with any existing register interface.
-Also note any differences.
-For example: Matches 16550 UART interface but registers are at 32-bit word offsets.
-```
-
-The next major section is a more detailed operational description of the module.
-
-```
-# Theory of Operations
-
-```
-
-Conventionally one of the first sections includes a block diagram and a description.
-
-_Should be useful to hardware designers, verification engineers and software engineers._
-
-
-```
-
-## Block Diagram
-
-![Name Block Diagram](block_diagram.svg)
-
-```
+## Line length and Whitespace
 
-There should be a section containing the automatically generated description of the IP including the signals, interrupts and alerts that it uses.
-
-_Primary user is the SoC integrator, but useful for everyone._
-
-Note that the interrupt descriptions are also automatically placed in the interrupt status register bit descriptions, which is the most likely place for software engineers to reference them.
-
-
-```
-
-## Hardware Interfaces
-
-
-```
-
-The organization of the design details section is done to suit the module.
-
-```
-
-## Design Details
-
-Details of the design.
-
-### Many third level headings
-```
-There are probably waveforms embedded here:
-
-```
-
-{{</* wavejson */>}}
-{
-  signal: [
-    { name: 'Clock',        wave: 'p............' },
-  ]
-}
-{{</* /wavejson */>}}
-
-```
-
-The final major section is the software user guide and describes using the IP and notes on writing device drivers.
-Code fragments are encouraged.
-
-_This section is primarily for software engineers, but it is expected that the code fragments are used by verification engineers._
-
-```
-
-# Programmers Guide
-
-```
-
-One important thing here is to show the order of initialization that has been tested in the verification environment.
-In most cases other orders will work, and may be needed by the software environment, but it can be helpful in tracking down bugs for the validated sequence to be described!
-
-```
-## Initialization
-```
-```c
+Files must have exactly one sentence per line, with no line breaks in the middle of a sentence.
+This way change reviews will highlight only those sentences which are modified.
+Though the long line lengths make the files slightly less convenient to read from the command-line, this greatly simplifies the review process.
+When reviewing Markdown changes, every altered sentence will be included in its entirety in the file diff.
 
- if (...) {
-   a = ...
- }
-```
+Markdown files must contain no trailing white space.
+As a result, two or more spaces must not be used for [hard line breaks][].
```

Comment:
Out of curiosity, what is this syntax:
```
[hard line breaks][]
```

Reply:
It's called a collapsed reference. `[hard line breaks][]` is the same as writing `[hard line breaks][hard line breaks]`.

See [the links section](https://github.com/HU90m/opentitan/blob/doc-docs/doc/contributing/style_guides/markdown_usage_style.md#links).